sparql-queries/sparql-queries.js

```diff
@@ -76,66 +99,183 @@ export async function finalizePublications(pubTask) {
 
     const now = new Date();
     const finishedStatus = 'http://themis.vlaanderen.be/id/concept/publication-task-status/success';
-    return await query(`
+    return await update(`
         ${PREFIXES}
-        
-        INSERT DATA {
-           GRAPH ${sparqlEscapeUri(pubTask.graph.value)} {
-                ${sparqlEscapeUri(pubTask.publicationTask.value)}       adms:status     ${sparqlEscapeUri(finishedStatus)} ;
-                                                                        dct:modified     ${sparqlEscapeDateTime(now)} .
-                
-           }
+        DELETE {
+            ${sparqlEscapeUri(pubTask.publicationTask)}       adms:status     ?oldStatus;
+                                                              dct:modified     ?oldDate.
+        }
+        INSERT {
+            ${sparqlEscapeUri(pubTask.publicationTask)}       adms:status     ${sparqlEscapeUri(finishedStatus)};
+                                                              dct:modified     ${sparqlEscapeDateTime(now)}.    
+        }
+        WHERE {
+            GRAPH ${sparqlEscapeUri(pubTask.graph)} {
+                ${sparqlEscapeUri(pubTask.publicationTask)}       a               ext:PublicationTask;      
+                                                                  adms:status     ?oldStatus;
+                                                                  dct:modified     ?oldDate .
+            }
         }
         
         `);
 }
 
 export async function saveHtmlContentToPublicationTask(pubTask, html) {
     // The HTML gets stored via nie:htmlContent to the publication task
-    const queryResult = await query(`
+    await update(`
     ${PREFIXES}
-    INSERT DATA {
-       GRAPH ${sparqlEscapeUri(pubTask.graph.value)} {
-            ${sparqlEscapeUri(pubTask.publicationTask.value)}        nie:htmlContent 	${sparqlEscapeString(html)} .
-       }
+    DELETE {
+        GRAPH ${sparqlEscapeUri(pubTask.graph)} {
+            ${sparqlEscapeUri(pubTask.publicationTask)}        nie:htmlContent 	?oldData.
+        }
+    }
+    INSERT {
+        GRAPH ${sparqlEscapeUri(pubTask.graph)} {
+            ${sparqlEscapeUri(pubTask.publicationTask)}        nie:htmlContent 	${sparqlEscapeString(html)}.
+        }
+    } 
+    WHERE {
+        GRAPH ${sparqlEscapeUri(pubTask.graph)} {
+            ${sparqlEscapeUri(pubTask.publicationTask)}        a                ext:PublicationTask;
+                                                               nie:htmlContent 	?oldData.
+        }
     }
     `);
-
-    return queryResult;
-
 }
 
 export async function pushEmailToOutbox(pubTask, html) {
     const now = new Date();
     const outbox = 'http://themis.vlaanderen.be/id/mail-folders/71f0e467-36ef-42cd-8764-5f7c5438ebcf';
-    const recipientBatches = createRecipientBatches(pubTask)
+    const recipientBatches = await createRecipientBatches(pubTask);
+    const attachmentsQuery = await generateAttachmentsQuery(pubTask);
```

Comment:
Ik vind het wat onlogisch om in de functie `pushEmailToOutbox` de attachments nog te gaan ophalen. Kan je dit niet beter op voorhand doen, zoals de `sources` en de attachments dan als extra argument meegeven aan de functie zodat de signature wordt `pushEmailToOutbox(pubTask, html, attachments)` waarbij attachments een lijst van file URIs is. In deze functie kan je dan de SPARQL query genereren. 

Meerdere values voor een predicaat kan je trouwens specifieren als volgt in een SPARQL query:
```
  <my-subject> <my-predicate> <value-1>, <value-2>, <value-3> .
```